pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -74,6 +74,9 @@ def backward(
         # enter apex context
         closure_loss = scaled_loss.__enter__()
 
+        # hook
+        model.trainer.call_hook("on_before_backward", closure_loss)
```

Comment:
is there no way to avoid calling `model.trainer` ?

Reply:
I don't think there is.

Reply:
yep, needs to call both the callbacks and the model hook